### ISSUE TITLE
Add globals option to ignore unfound dependencies

### ISSUE THREAD

**Issue by notadamking** [teambit/bit#1088]
**Requested feature**

The ability to define an array of project "globals" within the bit.json file. These globals would prevent missing dependencies errors for dependencies that match a path in the "globals" array.

**Motivation**

We have a component library that we'd like to share across repos, but in order to share that component, it would need to import different components with the same path based on which repo the component is in.

For example, we have a component library that depend on a UI component called a ContentSection. We'd like to now share that library with a separate repo, which also has a ContentSection, but the ContactSection in this repo is stylistically/functionally different than the ContentSection in the first repo, although it accepts the exact same props.

One way to solve this would be to always pass the ContentSection in as a prop to components in our component library, but that would require us to restructure our 80+ existing components. If we were instead able to list "/path-to-ContentSection" as a global dependency, we could share the component library with Bit and allow each repo to use a custom ContentSection component without changing every component in our library.

**Other use cases**

We would like to ignore other Bit component dependencies from our components, so that they don't get automatically bundled into a single src folder remotely.

**Comment by itaymendel** [comment]
(I just hate the fact that github on mobile does not allow me to tag people, I’ll edit it later and add some tags)

I see the reason why this is a required feature, as it will save a lot of refactoring, and still allow to use bit. However, just like ‘ignore-missing-deps’ flags, it is a kind of anti-pattern to be in a bit component.

For me it means that I want to be careful with how this is implemented. Ran&Gilad - we’ll need to see how it will affect the entire component-isolation for the project.

**Comment by adamjking3** [comment]
While I agree using this feature could be an anti-pattern, I would argue that for our team it is better. It would save us from needing to use the --ignore-missing-dependencies anti-pattern in multiple components, as well as enable our team to turn the component library mentioned above into a Bit component.

**Comment by itaymendel** [comment]
I know, and that’s why it is valid. As many teams face these issues. I want to figure out the safet way to go with it in bit.

**Comment by davidfirst** [comment]
@notadamking , This feature is now available (v14.1.0). The name is `overrides` (rather than `globals`).
A usage example:
```
{  
   "overrides":{  
      "ui/*":{  
         "dependencies":{  
            "file://./src/enums.js":"-"
         },
         "devDependencies":{  
            "@bit.utils/is-string":"-"
         },
         "peerDependencies":{  
            "react":"-"
         }
      }
   }
}
```
The above can be entered into `bit.json` file. It affects all components with "ui" namespace.
It's also possible to use a glob pattern with the files.

Please re-open if it doesn't address your use case.
